### PR TITLE
Birth module

### PR DESCRIPTION
**Pull request by TawanLekngam**

### CODE REVIEW

File: src/Pond.py [hunk 1]
```diff
@@ -1,38 +1,82 @@
...
 
-    def __init__(self, name: str, storage: Storage):
+    __UPDATE_EVENT = pygame.USEREVENT + 1
+    __PHEROMONE_EVENT = pygame.USEREVENT + 2
+
+    __BIRTH_RATE = 0.1
+
+    def __init__(self, name: str, storage: Storage = None):
         self.name: str = name
+        self.data: PondData = PondData(name)
```
Comment (deeckn) [inline src/Pond.py:26]: Could DI here I think.

File: src/Pond.py [hunk 2]
```diff
@@ -1,38 +1,82 @@
...
         self.storage: Storage = storage
+        self.fish_group = FishGroup()
```
Comment (deeckn) [inline src/Pond.py]: and DI here, your choice

File: src/Pond.py [hunk 3]
```diff
@@ -1,38 +1,82 @@
...
 
-from FishSprite import FishSprite
 from Storage import Storage
-from Dashboard import DashBoard
-from KeeperPanel import KeeperPanel
+from components import MainDashboard
+from models import FishSprite, FishGroup, PondData
+
+
+fish_factory = FishFactory()
```
Comment (deeckn) [inline src/Pond.py]: Could DI, if too much DI then no DI is ok too

File: src/Pond.py [hunk 4]
```diff
@@ -44,35 +88,37 @@ def run(self):
         update_time = pygame.time.get_ticks()
 
         self.load_fishes()
-        self.spawn_fish()
+
+        for _ in range(20):
```
Comment (deeckn) [inline src/Pond.py]: What is 20?

**Review by deeckn — COMMENTED** [review]
Looks good to me

**Review by deeckn — APPROVED** [review]

File: src/Pond.py (hunk 4 above)
Comment (TawanLekngam) [reply src/Pond.py]: Just for test

**Pull request merged**